src/ml_gym/util/util.py

```diff
@@ -61,6 +106,154 @@ def create_system_info() -> Dict:
         except Exception as e:
             raise SystemInfoFetchError(f"Unable to fetch System Info") from e
 
+@dataclass
+class TrainingDetails:
+    hyperparams: dict = None
+    loss_func: str = ""
+    optimizer: str = ""
+
+    def toJSON(self) -> Dict:
+        model_card ={}
+        model_card["hyperparams"] = self.hyperparams
+        model_card["loss_func"] = self.loss_func
+        model_card["optimizer"] = self.optimizer
+        return model_card
+
+@dataclass
+class EvalDetails:
+    loss_funcs: list = None
+    metrics: list = None
+
+    def toJSON(self) -> Dict:
+        model_card ={}
+        model_card["loss_funcs"] = self.loss_funcs
+        model_card["metrics"] = self.metrics
+        return model_card
+
+@dataclass
+class ModelCard:
+    model_details: ModelDetails
+    dataset_details: DatasetDetails
+    experiment_environment: ExperimentEnvironment
+    training_details: TrainingDetails
+    eval_details: EvalDetails
+
+    def toJSON(self) -> Dict:
+        model_card = {}
+        model_card["model_details"] = self.model_details.toJSON()
+        model_card["dataset_details"] = self.dataset_details.toJSON()
+        model_card["training_details"] = self.training_details.toJSON()
+        model_card["eval_details"] = self.eval_details.toJSON()
+        model_card["experiment_environment"] = self.experiment_environment.toJSON()
+        return model_card
+
+class SystemEnv:
```

Comment:
Should we rename this? 
ModelCardFactory?

Reply:
yes ofcourse